zulipterminal/core.py

```diff
@@ -273,16 +273,28 @@ def show_all_mentions(self, button: Any) -> None:
                         mentioned=True)
 
     def deregister_client(self) -> None:
-        queue_id = self.model.queue_id
-        self.client.deregister(queue_id, 1.0)
+        # Deregister before model has loaded.
+        if hasattr(self, 'model'):
+            queue_id = self.model.queue_id
+            self.client.deregister(queue_id, 1.0)
 
     def exit_handler(self, signum: int, frame: Any) -> None:
         self.deregister_client()
         sys.exit(0)
 
     @asynch
     def set_main_view(self) -> None:
+        def spinning_cursor() -> Any:
+            while True:
+                for cursor in '|/-\\':
+                    yield cursor
+
+        self.capture_stdout()
```

Comment:
Do we need to capture stdout here again? We already do this now in the previous commit in the `__init__` stage itself.